udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,17 +21,43 @@
 import java.util.Set;
 
 /**
- * Returns a collection of TimeRange objects to indicate available meeting times
- * for indicated people and duration given through a request object.
+ * Contains methods for querying availability between attendees.
  */
 public final class FindMeetingQuery {
-  private static final int OVERLAP_WITH_SAME_START = 1;
-  private static final int OVERLAP_WITH_SAME_END = 2;
-  private static final int OVERLAPS_END = 3;
-  private static final int OVERLAPS_START = 4;
-  private static final int OVERLAP_CONTAINS = 5;
+  private enum Overlap {
+    OVERLAP_WITH_SAME_START,
+    OVERLAP_WITH_SAME_END,
+    OVERLAPS_END,
+    OVERLAPS_START,
+    OVERLAP_CONTAINS,
+    NO_OVERLAP
+  }
 
+  /**
+   * Returns a collection of TimeRange objects to indicate available meeting times
+   * for indicated people and duration given through a request object. Determines
+   * whether or not to include optional attendees.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    // Perform queries with and without optional attendees.
+    Collection<TimeRange> queryWithoutOptionalAttendees = query(events, request, false);
+    Collection<TimeRange> queryIncludingOptionalAttendees = query(events, request, true);
```

Comment:
Can this be created in the `else` block below? This would avoid unnecessary computation if the collection above is returned.